### ISSUE TITLE
Right Click Functionality in web apps

### ISSUE THREAD

**Issue by liannacasper** [codenameone/CodenameOne#3247]
Right now it's problematic to use right click and "save link as" in a web app e.g. for Codename One Build. This makes features like "copy link" etc. problematic. It also means that things like build log etc always require a secondary click screen which ideally we'd like to avoid.

<img width="349" alt="Screen Shot 2020-08-30 at 14 36 24" src="https://user-images.githubusercontent.com/67953602/91658058-c0618300-eace-11ea-919f-90d1c2b29a87.png">

**Comment by shannah** [comment]
Was that "click to open link" from a recent build?  I introduced a number of measures a while back to avoid having to show that "Click to open link" dialog.

**Comment by liannacasper** [comment]
Yes but sometimes it doesn't happen so these fixes are working. However, we'd still need a way to right click on a "link" to save it/copy it like you would in a web browser. If not we'd probably need a way to do this with a dedicated button for copy to clipboard etc.

**Comment by shannah** [comment]
We could wrap the [clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard) to implement our existing copyToClipboard() method.  Doesn't look like this is implemented in JS port yet.

I have implemented copySelectionToClipboard(TextSelection) in the JS port - this can only be called from the native event thread so can only be triggered by a HeavyButton - which isn't ideal.  It was implemented before I developed the new methods (e.g. backside hooks) to avoid having to explicitly click on a native js component to trigger sensitive actions.
